gem/lib/rb_sys/mkmf.rb

```diff
@@ -34,23 +43,24 @@ def create_rust_makefile(target, cargo_dir = Dir.pwd)
         all: #{$extout ? "install" : "$(DLLIB)"}
       MAKE
 
-      File.write(File.join(cargo_dir, "Makefile"), make_install)
+      File.write("Makefile", make_install)
     end
     # rubocop:enable Style/GlobalVars
 
     private
 
     def base_makefile(cargo_dir)
-      base_makefile = dummy_makefile(cargo_dir).join("\n")
+      base_makefile = dummy_makefile(__dir__).join("\n")
       base_makefile.gsub!("all install static install-so install-rb", "all static install-so install-rb")
       base_makefile.gsub!("clean-so::", "clean-so:\n\t-$(Q)$(RM) $(DLLIB)\n")
+      base_makefile.gsub!(/^srcdir = .*$/, "srcdir = #{cargo_dir}")
       base_makefile
     end
 
     def cargo_command(cargo_dir, target)
       spec = Struct.new(:name, :metadata).new(target, {})
       builder = Gem::Ext::CargoBuilder.new(spec)
-      dest_path = File.join(cargo_dir, "target")
+      dest_path = File.join(Dir.pwd, "target")
       args = []
       cargo_cmd = builder.cargo_command(cargo_dir, dest_path, args)
       cargo_cmd.join(" ")
```

Comment:
Caused some annoying failures on mingw. can you give an example of the path that failed? was it before the `--` seperator?

Reply:
Ugh, I guess shellwords only does *nix style escaping.

I'm on macOS and have some stuff in Apple's iCloud Drive (kind of like Dropbox), the paths for the `--target-dir` and `--manifest-path` have spaces in them, like:

```
--target-dir /Users/mat/Library/Mobile Documents/com~apple~CloudDocs/src/halton-rb/tmp/arm64-darwin21/halton/3.1.0/target --manifest-path /Users/mat/Library/Mobile Documents/com~apple~CloudDocs/src/halton-rb/ext/halton/Cargo.toml
```

There's also some link args with paths, those don't have spaces for my setup, but it could be possible.

I had a quick try quoting the args, rather than escaping them, like:

``` ruby
cargo_cmd.map {|arg| "\"#{arg}\""}.join(" ")
```

and that works for me locally (and doesn't require loading shellwords as bonus).